includes/TripalFields/data__identifier/data__identifier_widget.inc

```diff
@@ -0,0 +1,259 @@
+
+    if ($element['#field_name'] == 'data__identifier' && strtolower($element['#title']) == 'identifier') {
+      // In identifier field.
+
+      // Text field. In either cases this field must be disabled from user.
+      $field_name = $this->field['field_name'];
+      $field_table = $this->instance['settings']['chado_table'];
+      $linker_field = 'chado-' . $field_table . '__uniquename';
+
+      // When interface to set prefix:
+      // Notice for admin where they can alter the prefix use to formulate uniquename.
+      // @see hook_menu().
+      $link = l('Update Prefix', 'admin/tripal/extension/trpfancy_fields/prefix');
+      tripal_set_message('Administrators, you can set prefix used in uniquename field in ' . $link . '.', TRIPAL_INFO, array('return_html' => FALSE));
+
+      // Dertermine if process is Create or Update. To test: check if chado_record_id
+      // were set to an id number.
+      if (isset($element['#entity']->chado_record_id) && $element['#entity']->chado_record_id > 0) {
+        // Update here.
+
+        $unique_name = array_key_exists($delta, $items) ? $items[$delta]['value'] : '';
+        $desciption = '';
+
+        // Feed to this field the uniquename for this stock.
+        $widget['value'] = array(
+          '#type' => 'value',
+          '#value' => array_key_exists($delta, $items) ? $items[$delta]['value'] : '',
+        );
+      }
+      else {
+        // Create.
+
+        // Pull system variable for configured prefix.
+        $prefix = variable_get('trpfancy_field_identifier_prefix');
+
+        if (isset($prefix) && !empty($prefix)) {
+          $unique_name = $prefix . ' + ID #';
+          $desciption = 'ID # will be set automatically on creation.';
+
+          // Use this to flag process that user is creating a new record
+          // and quick reference to the field.
+          $form['trpfancy_fields_field_name'] = array(
+            '#type' => 'value',
+            '#value' => $field_name,
+          );
+
+          $widget['table'] = array(
+            '#type' => 'value',
+            '#value' => $field_table,
+          );
+
+          $id = uniqid();
+          // What actually gets saved.
+          $widget['value'] = array(
+            '#type' => 'value',
+            '#value' => $id,
+          );
+        }
+      }
+
+      // The value presented to the user via load.
+      // If $items['delta']['value'] is set then we are updating and already have this
+      // information. As such, simply save it again.
+
+      // What user sees.
+      $widget[$linker_field] = array(
+        '#type'   => 'textfield',
+        '#title'   => 'Uniquename',
+        '#value'    => $unique_name,
+        '#disabled'  => TRUE,
+        '#description' => $desciption,
+      );
+    }
+    else {
+      // Text field but not identifier.
+      drupal_set_message('This option is available only to Identifier (text field).', 'error');
```

Comment:
It would be helpful if you added the name of the offending field here so administrators know which one to change.